IOPool/Streamer/src/EventMsgBuilder.cc

```diff
@@ -85,6 +93,12 @@ EventMsgBuilder::EventMsgBuilder(void* buf,
 
   event_addr_ = pos + sizeof(char_uint32);
   setEventLength(0);
+
+  if (headerSize() != expectedHeaderSize) {
```

Comment:
This check should make sure that the value computed by `computeHeaderSize(...)` is consistent with the header actually built.